contracts/route/route.go

```diff
@@ -23,6 +24,8 @@ type Route interface {
 	RunTLSWithCert(host, certFile, keyFile string) error
 	// ServeHTTP serves HTTP requests.
 	ServeHTTP(writer http.ResponseWriter, request *http.Request)
+	// Listen starts the HTTP server and listens for incoming connections on the specified listener.
+	Listen(l net.Listener) error
 	// Stop gracefully stop the serve.
 	Stop(ctx ...context.Context) error
```

Comment:
Q: I think it would be better to rename this method to Shutdown, which is what both the std library and go-fiber do.

Reply:
It was `Shutdown` previously, however, we are going to add `Stop` method to `Schedule`, `Queue`, and `Testing.Database`, so I changed it to `Stop`. But now, I think `Shutdown` is better, it means functions are closed gracefully. 👍 I'll create a PR for it.